diff --git a/scripts/Makefile.lib b/scripts/Makefile.lib
index b442d12ff2fe..e6453be3cacd 100644
--- a/scripts/Makefile.lib
+++ b/scripts/Makefile.lib
@@ -311,6 +311,11 @@ $(obj)/%.dtb.S: $(obj)/%.dtb
 
 quiet_cmd_dtc = DTC     $@
 cmd_dtc = mkdir -p $(dir ${dtc-tmp}) ; \
+	echo DTCDEBUG ; \
+	pwd ; \
+	ls -lh $(srctree) ; \
+	ls -lh $(srctree)/scripts/ ; \
+	file $(srctree)/scripts/dtc/dtc_overlay ; \
 	$(CPP) $(dtc_cpp_flags) -x assembler-with-cpp -o $(dtc-tmp) $< ; \
 	$(srctree)/scripts/dtc/dtc_overlay -@ -O dtb -o $@ -b 0 \
 		$(addprefix -i,$(dir $<) $(DTC_INCLUDE)) $(DTC_FLAGS) \
